.github/workflows/on_main_push.yml

```diff
@@ -0,0 +1,34 @@
+name: On Main Push
```

Comment:
I don't think we should have different sets for push and pull request. How would you know that the push is going to fail?